docs/templates.md

```diff
@@ -0,0 +1,175 @@
+sends and webpages it shows to users. Example templates can be found in
+the [res](https://github.com/matrix-org/sydent/tree/main/res) folder.
+
+## Branding
+
+Sending a value for `brand` to some API endpoints allows for different
+email and http templates to be used. These templates should be stored
+in a file structure like this:
+
+```
+ root_template_dir/
+     brand1/
+         [ brand1 template files ]
+     brand2/
+         [ brand2 template files ]
+```
+
+The config option `templates.root_directory` should be set to the path
+of `root_template_dir` and `templates.default_brand` should be set to
+the sub directory to use if no `brand` (or an invalid `brand`) is provided
+by the request.
+
+Here are the requests that can contain a value for `brand` and what
+template the brand value is used to select:
+
+Invite email templates:
+ - `POST /_matrix/identity/v2/validate/msisdn/requestToken`
+
+Verification email templates:
+ - `POST /_matrix/identity/v2/validate/email/requestToken`
+
+Verification SMS templates - BRAND CURRENTLY IGNORED:
+ - `POST /_matrix/identity/v2/validate/msisdn/requestToken`
+
+Verification response templates:
+ - `GET /_matrix/identity/v2/validate/email/submitToken`
+ - `GET /_matrix/identity/v2/validate/msisdn/submitToken`
+
+
+## Template formats
+
+### `.eml.j2` files
+
+Files ending in ".eml.j2" are Jinja templates. 
+
+Using the `urlencode` Jinja filter encodes the contents for URLs suitably.
+All variables are automatically made safe for HTML.
+
+(If needed the `safe` filter can be used to prevent the HTML encoding, i.e.
+declare the variable's contents "safe")
+
+See the Jinja [documentation](https://jinja.palletsprojects.com/en/3.0.x/templates/#variables)
+for more instructions on how to write these templates.
+
+### DEPRECATED `.eml` template files 
+
+Files ending in ".eml" can use python `%(variable)s` string substitution.
+
+Appending "_forurl" or "_forhtml" to any of the variable names listed below
+returns their values encoded suitably for URLs or HTML respectively.
+
+For example ">" in the HTML encoded version would be replaced with "&gt".
+
+---
+
+
+## Invite email templates 
+
+Invitation emails are sent when someone is invited to a room by their
+email address. They should contain a link for the user to click on
+that takes them to a matrix client.
+
+Invite template files should have the name `invite_template.eml.j2`.
+
+### Substitutions from Sydent
+
+Variable                | Contents 
+-----------             | -------- 
+`date`                  | The time and date of sending as defined in RFC 2822 (e.g. "Fri, 09 Nov 2001 01:08:47 -0000") 
+`ephemeral_private_key` | The ephemeral private key being used for this invite
+`from`                  | The sending email address as configured in `email.from`
+`messageid`             | The unique ID for this email 
+`multipart_boundary`    | Randomized multipart boundary to use in multipart emails
```

Comment:
Note that this doesn't have a `_forurl` or `_forhtml` substitute. It might be worth moving this out of the individual email sections and saying it is available for all emails?

Reply:
I've added a comment next to it saying that _forurl and _forhtml formats aren't available. I'm hesitant to put it somewhere else as I think it might be use ful to have all of the variables for each template grouped together (and so putting just one off on its own might be overlooked)